daemon/algod/api/server/v2/utils.go

```diff
@@ -578,6 +613,17 @@ func convertTxnGroupResult(txnGroupResult simulation.TxnGroupResult) PreEncodedS
 		UnnamedResourcesAccessed: convertUnnamedResourcesAccessed(txnGroupResult.UnnamedResourcesAccessed),
 	}
 
+	if len(txnGroupResult.PopulatedResourceArrays)-len(txnResults) > 0 {
```

Comment:
Is this really the right test?  Do you create an entry in PopulatedResourceArrays for non-app calls?  If not, then the length of the map might be 4, when the group had `[pay, pay, pay, app]`.  That means there are extras, you would need to pack `[pay, pay, pay, app, dummy app, dummy app, dummy app]` but this tests thinks everything's ok.